ci/scripts/retry

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+x() {
+    echo "+ $*" >&2
+    "$@"
+}
+
+max_retry_time_seconds=$(( 30 * 60 ))
+retry_delay_seconds=10
+
+END=$(( $(date +%s) + ${max_retry_time_seconds} ))
+
+while (( $(date +%s) < $END )); do
+    x "$@" && exit 0
+    sleep "${retry_delay_seconds}"
```

Comment:
but if there is a network causing CI to be flaky, so the retry will keep forever trying? 

I'm thinking we should have some cap, otherwise all pending jobs gonna dry out the resources

Reply:
you're right, there should be a cap
The script gives up after 30 minutes. Feels overly generous, but i just took whatever Trino uses today. We can obviously change the limit.

Reply:
I would put 3 mins as a cap. If there is an issue more than 3 mins it means the issue is severe on destination side and probably no sense to keep polling it increasing the traffic. 

Lets start with 3 mins and see how much failures we still have, we can adjust later

Reply:
I agree -- if we can't do network stuff in 3 minutes, continuing to retry is likely just going to hammer whatever system is having problems

Reply:
changed to 3m